nautobot/dcim/tests/test_api.py

```diff
@@ -2613,3 +2638,134 @@ def setUpTestData(cls):
+        cls.bulk_update_data = {
+            "software_version": software_versions[0].pk,
+            "status": statuses[0].pk,
+            "image_file_checksum": "abcdef1234567890",
+            "hashing_algorithm": SoftwareImageHashingAlgorithmChoices.SHA512,
+            "image_file_size": 1234567890,
+            "download_url": "https://example.com/software_image_test_case.bin",
+        }
+
+
+class SoftwareVersionTestCase(APIViewTestCases.APIViewTestCase):
+    model = SoftwareVersion
+
+    @classmethod
+    def setUpTestData(cls):
+        DeviceTypeToSoftwareImage.objects.all().delete()  # Protected FK to SoftwareImage prevents deletion
+        cls.VERBOTEN_STRINGS = tuple(
+            [o for o in cls.VERBOTEN_STRINGS if o not in SoftwareImageHashingAlgorithmChoices.as_dict().keys()]
+        )
+        statuses = Status.objects.get_for_model(SoftwareVersion)
+        platforms = Platform.objects.all()
+
+        cls.create_data = [
+            {
+                "platform": platforms[0].pk,
+                "status": statuses[0].pk,
+                "version": "version 1.1.0",
+            },
+            {
+                "platform": platforms[1].pk,
+                "status": statuses[1].pk,
+                "version": "version 1.2.0",
+            },
+            {
+                "platform": platforms[2].pk,
+                "status": statuses[2].pk,
+                "version": "version 1.3.0",
+            },
+        ]
+        cls.bulk_update_data = {
+            "platform": platforms[0].pk,
+            "status": statuses[0].pk,
+            "alias": "Version x.y.z",
+            "release_date": datetime.date(2001, 12, 31),
+            "end_of_support_date": datetime.date(2005, 12, 31),
+            "documentation_url": "https://example.com/software_version_test_case/docs2",
+            "long_term_support": False,
+            "pre_release": True,
+        }
+
+
+class DeviceTypeToSoftwareImageTestCase(APIViewTestCases.APIViewTestCase):
+    model = DeviceTypeToSoftwareImage
+
+    @classmethod
+    def setUpTestData(cls):
+        DeviceTypeToSoftwareImage.objects.all().delete()
+        cls.VERBOTEN_STRINGS = tuple(
+            [o for o in cls.VERBOTEN_STRINGS if o not in SoftwareImageHashingAlgorithmChoices.as_dict().keys()]
+        )
+        device_types = DeviceType.objects.all()[:4]
+        software_images = SoftwareImage.objects.all()[:3]
+
+        # deletable objects
+        DeviceTypeToSoftwareImage.objects.create(
+            device_type=device_types[0],
+            software_image=software_images[0],
+            is_default=True,
+        )
+        DeviceTypeToSoftwareImage.objects.create(
+            device_type=device_types[0],
+            software_image=software_images[1],
+            is_default=False,
+        )
+        DeviceTypeToSoftwareImage.objects.create(
+            device_type=device_types[0],
+            software_image=software_images[2],
+            is_default=True,
+        )
```

Comment:
Seems like we are allowing two `is_default=True` software images on the same device type here.